varnish-macros/src/stats.rs

```diff
@@ -0,0 +1,226 @@
+use crate::parser_utils::{find_attr, has_attr, parse_doc_str};
+use serde::Serialize;
+use std::collections::HashMap;
+use std::mem::size_of;
+use std::sync::atomic::AtomicU64;
+use syn::{punctuated::Punctuated, token::Comma, Data, Field, Fields, Type};
+
+type FieldList = Punctuated<Field, Comma>;
+
+#[derive(Serialize, Clone)]
+enum CType {
+    #[serde(rename = "uint64_t")]
+    Uint64,
+}
+
+impl CType {
+    const fn size(&self) -> usize {
+        match self {
+            CType::Uint64 => size_of::<AtomicU64>(),
+        }
+    }
+}
+
+#[derive(Serialize, Clone)]
+#[serde(rename_all = "lowercase")]
+enum MetricType {
+    Counter,
+    Gauge,
+}
+
+#[derive(Serialize, Clone)]
+#[serde(rename_all = "lowercase")]
+enum Level {
+    Info,
+    Diag,
+    Debug,
+}
+
+impl Default for Level {
+    fn default() -> Self {
+        Self::Info
+    }
+}
+
+#[derive(Serialize, Clone)]
+#[serde(rename_all = "lowercase")]
+enum Format {
+    Integer,
+    Bitmap,
+    Duration,
+    Bytes,
+}
+
+impl Default for Format {
```

Comment:
you can do this by simply adding `#[default]` next to `Integer` in the enum, and adding `Default` to the derive attribute